File: memories/2024/exercise/ts/day10/src/delivery/building3.ts
export class Building3 {
    static whichFloor(instructions: string): number {
        const hasElf = instructions.includes("🧝");
        let val: number[] = [];
        
        for (let i = 0; i < instructions.length; i++) {
            const c = instructions[i];
            
            if (hasElf) {
                val.push(c === ')' ? 3 : -2);
            } else {
                val.push(c === '(' ? 1 : -1);
            }
        }
        
        let result = 0;
        for (const v of val) {
            result += v;
        }
        return result;
    }
}

File: memories/2024/exercise/kotlin/day10/src/main/kotlin/delivery/Building3.kt
package delivery

object Building3 {
    fun whichFloor(instructions: String): Int {
        val hasElf = instructions.contains("🧝")
        val valList = mutableListOf<Int>()
        
        for (i in instructions.indices) {
            val c = instructions[i]
            
            if (hasElf) {
                valList.add(if (c == ')') 3 else -2)
            } else {
                valList.add(if (c == '(') 1 else -1)
            }
        }
        
        var result = 0
        for (v in valList) {
            result += v
        }
        return result
    }
}